crud/select/plan.lua

```diff
@@ -48,49 +48,75 @@ local function get_index_for_condition(space_indexes, space_format, condition)
     end
 end
 
-local function extract_sharding_key_from_scan_value(scan_value, scan_index, sharding_index)
-    if #scan_value < #sharding_index.parts then
+local function extract_sharding_key_from_conditions(conditions, sharding_index, space_indexes, fieldno_map)
```

Comment:
Add `dev_checks()` in `extract_sharding_key_from_conditions()`.

Reply:
Added